datashader/spatial/__init__.py

```diff
@@ -391,182 +433,210 @@ def query_partitions(self, x_range, y_range):
             Dask dataframe containing all data from all partitions that
             potentially intersect with the specified x_range/y_range box.
         """
+        if self.spatial is None:
+            raise RuntimeError("SpatialPointFrame is missing spatial "
```

Comment:
>Are there legitimate cases where this would come up in practice?

I don't know, I haven't exhaustively tested every dask method to see if any of them make a new SpatialDataFrame instance. The other case where this may come up is if a user tries to construct one of these directly. It just seems safer to do this.

>If so, should this be a warning (of decreased performance) rather than an error?

``canvas.points`` already checks if the object has the spatial namespace, so the only case where this would occur is if a user tries to call this method directly in which case this absolutely **should** error.

Reply:
Ok, sounds good.